### PR TITLE
[JENKINS-62910] Table to div compatibility

### PR DESCRIPTION
**Pull request by res0nance**
Starting a draft PR first, mostly to make sure its working with CI / incrementals

### CODE REVIEW

**Comment by res0nance** [comment]
AFAICT I can't seem to break this, the configuration saves correctly.

CC: @fqueiruga , @timja In case you guys want to look, but I cannot seem to cause the typical issues with the latest version of Jenkins. I've updated the POM for convenience.

File: pom.xml [hunk 1]
```diff
@@ -6,28 +6,31 @@
...
   </scm>
 	<properties>
+		<revision>1.3.7</revision>
+		<changelist>-SNAPSHOT</changelist>
+		<gitHubRepo>jenkinsci/conditional-buildstep-plugin</gitHubRepo>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<maven.javadoc.skip>true</maven.javadoc.skip>
-		<jenkins.version>1.580.1</jenkins.version>
-		<java.level>6</java.level>
+		<jenkins.version>2.176.4</jenkins.version>
```
Comment (timja) [inline pom.xml:32]: best to go to at least 2.204.6 if not later, that means there's no implied dependencies on detached plugins

**Comment by timja** [comment]
> AFAICT I can't seem to break this, the configuration saves correctly.
> 
> CC: @fqueiruga , @timja In case you guys want to look, but I cannot seem to cause the typical issues with the latest version of Jenkins. I've updated the POM for convenience.

only certain cases break it, normally to do with nested table tags or repeatable elements.

It's hard to tell what will break it too.

**Comment by res0nance** [comment]
> > AFAICT I can't seem to break this, the configuration saves correctly.
> > CC: @fqueiruga , @timja In case you guys want to look, but I cannot seem to cause the typical issues with the latest version of Jenkins. I've updated the POM for convenience.
> 
> only certain cases break it, normally to do with nested table tags or repeatable elements.
> 
> It's hard to tell what will break it too.

If its not going to break in this case, we should just migrate from a table to a div then. WDYT?

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
> > > AFAICT I can't seem to break this, the configuration saves correctly.
> > > CC: @fqueiruga , @timja In case you guys want to look, but I cannot seem to cause the typical issues with the latest version of Jenkins. I've updated the POM for convenience.
> > 
> > 
> > only certain cases break it, normally to do with nested table tags or repeatable elements.
> > It's hard to tell what will break it too.
> 
> If its not going to break in this case, we should just migrate from a table to a div then. WDYT?

only if you bump the baseline to 2.265 you can do that.

**Review by timja — APPROVED** [review]

**Comment by res0nance** [comment]
Confirming that there is a case where it breaks, thanks to a user for providing a reproducible example.

CC: @oleg-nenashev IIRC has permissions to release this.

**Comment by oleg-nenashev** [comment]
Yes. This issue is one of the reasons why I asked for permissions. Will review/merge shortly

**Review by oleg-nenashev — APPROVED** [review]
If there is any concern, I suggest that we cut a compatible version for 2.204 and then, if things still fail for 2.265, work on a new release for 2.265+ which switches to the new config entirely

**Comment by oleg-nenashev** [comment]
Will merge once green

**Comment by res0nance** [comment]
> If there is any concern, I suggest that we cut a compatible version for 2.204 and then, if things still fail for 2.265, work on a new release for 2.265+ which switches to the new config entirely

A user reported that he tested the new version with 2.266, so I don't anticipate issues but in case there is the path forward is clear.

**Pull request merged**